checker/src/org/checkerframework/checker/lock/LockVisitor.java

```diff
@@ -4,89 +4,369 @@
+     * and the receiver actual parameter has type @GuardedBy(...), this method verifies that
+     * the guard is satisfied, and it returns true, indicating that the receiver subtype check should be skipped.
+     * If the receiver actual parameter has type @GuardSatisfied, this method simply returns true without
+     * performing any other actions. The method returns false otherwise.
+     *
+     * @param node the MethodInvocationTree of the method being called.
+     * @param methodDefinitionReceiver the ATM of the formal receiver parameter of the method being called.
+     * @param methodCallReceiver the ATM of the receiver argument of the method call.
+     * @return whether the caller can skip the receiver subtype check.
+     */
+    @Override
+    protected boolean skipReceiverSubtypeCheck(MethodInvocationTree node,
+            AnnotatedTypeMirror methodDefinitionReceiver,
+            AnnotatedTypeMirror methodCallReceiver) {
+
+        AnnotationMirror primaryGb = methodCallReceiver.getAnnotationInHierarchy(GUARDEDBYUNKNOWN);
+        AnnotationMirror effectiveGb = methodCallReceiver.getEffectiveAnnotationInHierarchy(GUARDEDBYUNKNOWN);
+
+        // If the receiver actual parameter has type @GuardSatisfied, skip the subtype check.
+        // Consider only a @GuardSatisfied primary annotation - hence use primaryGb instead of effectiveGb.
+        if (primaryGb != null && AnnotationUtils.areSameByClass(primaryGb, checkerGuardSatisfiedClass)){
+            AnnotationMirror primaryGbOnMethodDefinition = methodDefinitionReceiver.getAnnotationInHierarchy(GUARDEDBYUNKNOWN);
+            if (primaryGbOnMethodDefinition != null && AnnotationUtils.areSameByClass(primaryGbOnMethodDefinition, checkerGuardSatisfiedClass)) {
+                return true;
+            }
+        }
+
+        if (AnnotationUtils.areSameByClass(effectiveGb, checkerGuardedByClass)) {
+            Set<AnnotationMirror> annos = methodDefinitionReceiver.getAnnotations();
+            for (AnnotationMirror anno : annos) {
+                if (AnnotationUtils.areSameByClass(anno, checkerGuardSatisfiedClass)) {
+                    MethodInvocationNode methodInvocationNode = (MethodInvocationNode) atypeFactory.getNodeForTree(node);
+                    Node receiverNode = methodInvocationNode.getTarget().getReceiver();
+                    checkPreconditions(node, receiverNode,
+                            generatePreconditionsBasedOnGuards(methodCallReceiver));
+
+                    return true;
+                }
+            }
+        }
+
+        return false;
+    }
+
+    @Override
+    protected Set<? extends AnnotationMirror> getExceptionParameterLowerBoundAnnotations() {
+        Set<? extends AnnotationMirror> tops = atypeFactory.getQualifierHierarchy().getTopAnnotations();
+        Set<AnnotationMirror> annotationSet = AnnotationUtils.createAnnotationSet();
+        for (AnnotationMirror anno : tops) {
+            if (anno.equals(GUARDEDBYUNKNOWN)) {
```

Comment:
To follow up on this comment, I filed issue #658. Unfortunately, because there is no toString Element in AnnotationMirror, we currently can't use the Report Checker.